### PR TITLE
[Feat-16] - Scraping stats

### LINKED ISSUE #16: Provide stats for scraping data
### Acceptance
- [x] Total No. of products scraped
- [x] Elapsed time
- [x] Products scraper per second
- [x] Aprox. Time to scrape 1 Million products
- [x] Current concurrency
- [x] Avg. Response API call time
- [x] Completed categories with product count, elapsed time, no. of pages
- [x] Ongoing categories with product count
- [x] Should have a option to get all the stats as numbers otherwise human readable format like (3.1K, 1H 20M)

### PR DESCRIPTION
**Pull request by asrafalih**
### Acceptance
- [x] Total No. of products scraped
- [x] Elapsed time
- [x] Products scraper per second
- [x] Aprox. Time to scrape 1 Million products
- [x] Current concurrency
- [x] Avg. Response API call time
- [x] Completed categories with product count, elapsed time, no. of pages
- [x] Ongoing categories with product count
- [x] Should have a option to get all the stats as numbers otherwise human readable format like (3.1K, 1H 20M)
- [x] TypeDoc generation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,4 +69,4 @@
     "url": "https://github.com/hi-imcodeman/flipkart-scraper/issues"
   },
   "homepage": "https://hi-imcodeman.github.io/flipkart-scraper"
-}
\ No newline at end of file
+}
```
Comment (athena2207) [inline package.json:72]: add a new line

File: src/enums.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export enum ScraperStatus {
+    READY = 'ready',
+    INPROGRESS = 'inprogress',
+    PAUSED = 'paused',
+    STOPPED = 'stopped',
+    FINISHED = 'finished'
+}
```
Comment (athena2207) [inline src/enums.ts]: add a new line

File: src/util.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
-export const sleep = (ms: number): Promise<void> => {
+import numeral from 'numeral'
+
+export const sleep = async (ms: number): Promise<void> => {
     return new Promise(resolve => {
         setTimeout(() => {
             resolve()
         }, ms)
     })
-}
```
Comment (athena2207) [inline src/util.ts:7]: add new line

File: tsconfig.json [hunk 4]
```diff
@@ -19,7 +19,9 @@
...
     "excludePrivate": true,
     "excludeExternals": true,
-    "entryPoints": ["./src/index.ts"],
+    "excludeInternal": true,
+    "exclude": ["**/*+(util|.spec|.e2e).ts","**/__mocks__"],
+    "entryPoints": ["src"],
     "out": "docs"
   }
-}
+}
```
Comment (athena2207) [inline tsconfig.json]: add new line

File: src/util.ts [hunk 5]
```diff
@@ -1,7 +1,11 @@
...
\ No newline at end of file
+}
+
+export const durationFormatWithMs = (ms: number):string => `${numeral(ms / 1000).format('00:00:00')} ${ms % 1000}ms`
```
Comment (athena2207) [inline src/util.ts]: add new line

File: package.json (hunk 1 above)
Comment (asrafalih) [reply package.json:72]: > add a new line

Already it's there

File: src/enums.ts (hunk 2 above)
Comment (asrafalih) [reply src/enums.ts]: done

File: src/util.ts (hunk 3 above)
Comment (asrafalih) [reply src/util.ts:7]: done

File: tsconfig.json (hunk 4 above)
Comment (asrafalih) [reply tsconfig.json]: done

File: src/util.ts (hunk 5 above)
Comment (asrafalih) [reply src/util.ts]: done

**Pull request merged**